### PR TITLE
Create solution

### PR DESCRIPTION
**Pull request by ivan-kadykalo**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,81 @@
...
+      throw new Error('Invalid data input');
+    }
+
+    if (this.c >= this.a + this.b) {
+      throw new Error('sides 1, 2 and 3 can not form a triangle');
+    }
+  }
+
+  getArea(): number {
+    const s = ((this.a + this.b + this.c) / 2);
```
Comment (YegorKochetkov) [inline src/figuresClasses.ts:34]: non-informative variable

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,81 @@
...
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Invalid data input');
+    }
+
+    if (this.c >= this.a + this.b) {
```
Comment (YegorKochetkov) [inline src/figuresClasses.ts:28]: what happens if the hypotenuse is passed as the first or second argument?

**Review by YegorKochetkov — CHANGES_REQUESTED** [review]
minor flaws)

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,85 @@
...
+  getArea(): number {
+    return Math.floor((Math.PI * (this.radius * this.radius)) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shapes = Shapes.Rectangle;
+
+  area: number = 0;
```
Comment (AlinaMatuschak) [inline src/figuresClasses.ts:66]: Is necessary to declare area here?

**Review by AlinaMatuschak — APPROVED** [review]

**Pull request closed without merging**